File: Data Structures & Algorithms/top-k-elements-in-list/submission-16.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        List<Integer> [] frequency = new List[nums.length + 1];
        Map<Integer, Integer> count = new HashMap<>();

        for(int i = 0; i < frequency.length; i++){
            frequency[i] = new ArrayList<>();
        }

        for(int current: nums){
            count.put(current, count.getOrDefault(current, 0) + 1);
        }

        for(Map.Entry<Integer, Integer> current : count.entrySet()){
            frequency[current.getValue()].add(current.getKey());
        }

        int [] result = new int [k];
        int index = 0;

        for(int i = frequency.length -1; i > 0; i--){
            for(int c: frequency[i]){
                result[index++] = c;


                if(index == k){
                    return result;
                }  
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-16.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        values = set()
        maxSequence = 0

        for num in nums:
            values.add(num)

        for value in values:
            current = 1

            if value - 1 not in values:
                value = value + 1
                while value in values:
                    current += 1
                    value += 1
            
            maxSequence = max(current, maxSequence)
        
        return maxSequence

